### PR TITLE
FIX(plugins): Load correct pages for modules

### LINKED ISSUE #6558: Mumble lags and freezing when Left 4 Dead 2 is launched. PA not working
### Description

So I was trying to use positional audio in Left 4 Dead 2 (2.2.4.1). "Link to game and transmit position", "enable" and "PA" for Source Engine is ticked, but Mumble is getting super laggy and freezing every 2 seconds for me and my friend when the game is launched. Like voice is cracking and even the windows of the program freeze when you move them. Despite this, we checked if the positional audio even works and no, it's not working. We hear each other like in a normal chat (with voice cracking and huge latency ofc huh)

P.S. I am using Mumble for the first time, and there are no specific guides for L4D2.

### Steps to reproduce

1. Tick "Link to game and transmit position", "enable" and "PA" for Source Engine
2. Launch Left 4 Dead 2

### Mumble version

1.5.634

### Mumble component

Client

### OS

Windows

### Reproducible?

Yes

### Additional information

_No response_

### Relevant log output

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by sqwishy**
This `VirtualQueryEx()` loop is called for each module in a process. It reads pages starting at the module address but seems to continue past into other modules and into dynamic allocations also.

This check stops enumerating pages once it encounters one that no longer belongs to the module for which pages are being collected.

(Also this function opens two handles, this adds a clean up for the first handle if opening the second fails.)

Fixes #6558

### Checks

- [X] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

File: plugins/HostWindows.cpp [hunk 1]
```diff
@@ -49,7 +50,8 @@ Modules HostWindows::modules() const {
 		MEMORY_BASIC_INFORMATION64 mbi;
 		auto address = reinterpret_cast< procptr_t >(me.modBaseAddr);
 		while (VirtualQueryEx(processHandle, reinterpret_cast< LPCVOID >(address),
-							  reinterpret_cast< PMEMORY_BASIC_INFORMATION >(&mbi), sizeof(mbi))) {
+							  reinterpret_cast< PMEMORY_BASIC_INFORMATION >(&mbi), sizeof(mbi))
+			   && (mbi.AllocationBase == reinterpret_cast< procptr_t >(me.modBaseAddr))) {
```
Comment (Krzmbrzl) [inline plugins/HostWindows.cpp:54]: There should be a comment explaining what is done here and why. That way future coders have an easier time understanding what's happening :)

**Comment by davidebeatrici** [comment]
Oh... now I feel bad for complaining about the API in #6491 and #6558 as there is a proper reason for performance to be so terrible.

It's entirely my fault, as I'm the sole person who worked on the code. Sorry about that.

**Review by davidebeatrici — APPROVED** [review]

**Comment by davidebeatrici** [comment]
Thank you for finding the cause and submitting a fix!

**Comment by Krzmbrzl** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|1.5.x|[](https://github.com/mumble-voip/mumble/pull/6600)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport) and see the [Github Action logs](https://github.com/mumble-voip/mumble/actions/runs/11186834009) for details

**Pull request merged**